mcsema/Arch/ABI.cpp

```diff
@@ -518,6 +545,36 @@ static const char *GetVarImpl(
   return nullptr;
 }
 
+static void StorePackedType(llvm::BasicBlock *block, llvm::Value *ret_val) {
+  const auto &regs = CallingConventionInfo::GetPackedReturn(
+      KindOfValue(ret_val->getType()));
+
+  CHECK(!regs.names.empty() || regs.accepted_kinds == kInvalidKind)
+      << "Could not store return type "
+      << remill::LLVMThingToString(ret_val->getType());
+
+  llvm::IRBuilder<> ir(block);
+
+  // Need to get ptr to some type smaller than actual type is, as
+  // it will be split into multiple registers
+  auto ret_val_alloca = ir.CreateAlloca(ret_val->getType());
+  ir.CreateStore(ret_val, ret_val_alloca);
+  auto ptr_ret_val = ir.CreateBitCast(ret_val_alloca, regs.unit_ptr_type);
+
+  for (auto i = 0U; i < regs.names.size(); ++i) {
+    // Get partial value from offset
+    auto partial_value_ptr = ir.CreateGEP(ptr_ret_val, GetConstantInt(64, i));
+    auto partial_value = ir.CreateLoad(partial_value_ptr);
+
+    llvm::Value *dest_loc = remill::FindVarInFunction(block, regs.names[i]);
+
+    // Store actual value
+    dest_loc = ir.CreateBitCast(dest_loc,
+                                regs.unit_ptr_type);
+    ir.CreateStore(partial_value, dest_loc);
```

Comment:
I think you want to mark these registers as now being used, or is that not necessary?

Reply:
It is a correct thing to do. However I don't think it's necessary, since I was not able to simulate return that is packed like this + returns something else in registers ( whole type gets `sret` instead).
There is also a check for this before entering L873 ` CHECK(GetNumberOfElements(val_type) == 1)` so if it actually happens check fails and appropriate error is printed.